extensions-contrib/kafka-emitter/src/main/java/org/apache/druid/emitter/kafka/KafkaEmitterConfig.java

```diff
@@ -23,22 +23,49 @@
 import com.fasterxml.jackson.annotation.JsonProperty;
 import com.google.common.base.Preconditions;
 import com.google.common.collect.ImmutableMap;
+import org.apache.commons.lang3.EnumUtils;
 import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.common.errors.InvalidConfigurationException;
 
 import javax.annotation.Nullable;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
 import java.util.Map;
+import java.util.Set;
 
 public class KafkaEmitterConfig
 {
+  public enum EventType
+  {
+    metrics,
+    alerts,
+    requests,
+    segment_metadata
+  }
+
+  public static final Set<EventType> DEFAULT_EVENT_TYPES = new HashSet<>(Arrays.asList(EventType.alerts, EventType.metrics));
+
+  public enum SegmentMetadataTopicFormat
+  {
+    json,
+    protobuf
+  }
 
   @JsonProperty(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG)
   private final String bootstrapServers;
-  @JsonProperty("metric.topic")
+  @Nullable @JsonProperty("event.types")
+  private Set<EventType> eventTypes;
+  @Nullable @JsonProperty("metric.topic")
   private final String metricTopic;
-  @JsonProperty("alert.topic")
+  @Nullable @JsonProperty("alert.topic")
   private final String alertTopic;
   @Nullable @JsonProperty("request.topic")
   private final String requestTopic;
+  @Nullable @JsonProperty("segment.metadata.topic")
+  private String segmentMetadataTopic;
+  @Nullable @JsonProperty("segment.metadata.topic.format")
+  private SegmentMetadataTopicFormat segmentMetadataTopicFormat;
```

Comment:
we should name those configurations `segmentMetadata.topic.xxxx` to align all the configs to follow the same `<eventType>.topic.format` hierarchy, using the same names we define in the event type config (using cameCase as mentioned above to align with existing conventions.)